### ISSUE TITLE
S3CopyToTable.s3_load_path TypeError

### ISSUE THREAD

**Issue by kianho** [spotify/luigi#995]
I encountered this `TypeError` when subclassing `S3CopyToTable`:
```
...
Traceback (most recent call last):
  File "/home/kian/workspaces/contrib/luigi/luigi/worker.py", line 137, in run
    new_deps = self._run_get_new_deps()
  File "/home/kian/workspaces/contrib/luigi/luigi/worker.py", line 88, in _run_get_new_deps
    task_gen = self.task.run()
  File "/home/kian/workspaces/contrib/luigi/luigi/contrib/redshift.py", line 166, in run
    path = self.s3_load_path()
TypeError: 'str' object is not callable
INFO: Skipping error email. Set `error-email` in the `core` section of the luigi config file to receive error emails.
DEBUG: 1 running tasks, waiting for next task to finish
DEBUG: Asking scheduler for work...
INFO: Done
INFO: There are no more tasks to run at this time
INFO: Worker Worker(salt=532581476, workers=1, host=..., username=kian, pid=23435) was stopped. Shutting down Keep-Alive thread
```
which was fixed by changing https://github.com/spotify/luigi/blob/master/luigi/contrib/redshift.py#L166 from:
```python
path = self.s3_load_path()
``` 
to
```python
path = self.s3_load_path
```
(__I submitted this fix as PR #996__) 

as per the other class properties, which as far as i can tell, aren't explicitly referenced as methods either (otherwise they too would raise the above exception?). 

The snippet I used to generate the above error:
```python
import luigi
    
from luigi.s3 import S3Target, S3Client                                                                          
from luigi.contrib.redshift import S3CopyToTable                                                                 
    
    
class MyS3Task(luigi.Task): 
    local_tsv = luigi.Parameter()
    s3_load_path = luigi.Parameter()                                                                             
    client = luigi.Parameter()
    
    def output(self):
        return S3Target(self.s3_load_path, client=self.client)

    def run(self):
        self.client.put(self.local_tsv, self.output().path)
        return


class MyRedshiftTask(S3CopyToTable):
    host = luigi.Parameter()
    database = luigi.Parameter()
    user = luigi.Parameter()
    password = luigi.Parameter()
    table = luigi.Parameter()
    local_tsv = luigi.Parameter()

    aws_access_key_id = luigi.Parameter()
    aws_secret_access_key = luigi.Parameter()

    columns = [("x", "INT"),
               ("y", "INT")]

    s3_load_path = luigi.Parameter()
    copy_options = "IGNOREHEADER 1"

    def requires(self):
        client = S3Client(self.aws_access_key_id, self.aws_secret_access_key)
        return MyS3Task(s3_load_path=self.s3_load_path,
                        local_tsv=self.local_tsv, client=client)


if __name__ == '__main__':
    luigi.run()
```
which was run from the command line using:
```bash
python ./luigi_example.py --local-scheduler MyRedshiftTask \
    --host "<REDSHIFT ENDPOINT>:5439" \
    --database "dev" \
    --user "<USERNAME>" \
    --password "<HIDDEN>" \
    --table "test_redshift_table_5439" \
    --aws-access-key-id "<HIDDEN>" \
    --aws-secret-access-key "<HIDDEN>" \
    --s3-load-path "s3://bucket-5439/test.tsv" \
    --local-tsv "./test.tsv" \
```
where `test.tsv` (tab-separated values) contained:
```
x   y
1   2
10  20
100 200
```

**Comment by Tarrasch** [comment]
Do `git blame` to find the authors and @-mention them. :)

**Comment by kianho** [comment]
ah yes of course, @ddaniels888 @rantav 
thanks!

**Comment by carlineng** [comment]
This commit has broken a few of my S3CopyToTable tasks. In my tasks, the S3 path is dynamically determined at runtime, rather than passed in as a parameter to the job, so I need to override the `s3_load_path`, rather than define it as a class property.

The above case for which this PR was initially introduced could be fixed by something like this:

{code}
class MyRedshiftTask(S3CopyToTable):
    host = luigi.Parameter()
    database = luigi.Parameter()
    user = luigi.Parameter()
    password = luigi.Parameter()
    # ...

    load_path_param = luigi.Parameter()

    def s3_load_path(self):
        return self.load_path_param
{code}

If the S3CopyToTask is grabbing the `s3_load_path` value as a class property, then there's no way to define it dynamically.

**Comment by Tarrasch** [comment]
@carlineng, this is a issue not a PR. You should also at-mention the author while you're at it. :)

**Comment by kianho** [comment]
Thanks for raising this @carlineng, that is indeed a more appropriate fix, I'll create a new PR to address this.

**Comment by kianho** [comment]
@carlineng, issue addressed with #1096

**Comment by dlstadther** [comment]
Since a PR was submitted and merged to correct this, can we close? Thanks!
